### PR TITLE
netutils server

### PR DESCRIPTION
**Pull request by rajatchopra**
An http REST based IPAM server - basic implementation that serves IPs for now, but meant to serve subnets too.
The test uses the netutils package as an example.

@mrunalp PTAL

### CODE REVIEW

File: pkg/netutils/server/server.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	w.Header().Add("Content-type", "application/json")
+	w.Write([]byte("Not implemented"))
+	return
+}
+
+// handleIP handles IP requests
+func (s *Server) handleIP(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "GET" {
+		w.Header().Add("Content-type", "application/json")
+		ipnet, _ := s.ipam.GetIP()
```
Comment (mrunalp) [inline pkg/netutils/server/server.go]: Error checking here?

File: pkg/netutils/server/server_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	if err != nil {
+		t.Fatal("Error in obtaining IP address through server")
+	}
+	res.Body.Close()
+	return string(ip)
+}
+
+func TestIPServe(t *testing.T) {
+	inuse := make([]string, 0)
+	ipam, _ := netutils.NewIPAllocator("10.20.30.40/24", inuse)
```
Comment (mrunalp) [inline pkg/netutils/server/server_test.go]: error check

File: pkg/netutils/server/server.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	}
+	if tlsOptions != nil {
+		s.TLSConfig = tlsOptions.Config
+		s.ListenAndServeTLS(tlsOptions.CertFile, tlsOptions.KeyFile)
+	} else {
+		s.ListenAndServe()
+	}
+}
+
+// IpamInterface contains all the methods required by the server.
```
Comment (mrunalp) [inline pkg/netutils/server/server.go]: Maybe move this up above the Server struct?

**Comment by rajatchopra** [comment]
Fixed.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**